### ISSUE TITLE
./configure $'\r': command not found

### ISSUE THREAD

**Issue by K0rrid0r** [RTimothyEdwards/irsim#2]
![image](https://user-images.githubusercontent.com/56724286/113068206-1fe75480-91be-11eb-8cfd-a098c72c1f42.png)

I've tried for hours now. I cant get it to work, or am I missing something? some extra lib?

Irsim is cloned into C:\cygwin64\usr\local\src
from there i do ./configure

./configure --disable-usersubckt
gives the same error as well

I have Magic working from the same method so i really don't understand this :)

I'm on Windows 8.1

**Comment by RTimothyEdwards** [comment]
I'm a bit weirded out by the comment about the dollar sign followed by the carriage return.  It sounds like the "configure" script has got a break after the $ that shouldn't be there.  Line 7 of "configure" should be

`( CFLAGS="-g" ; export CFLAGS ; cd scripts ; ./configure "$@" )`

all on one line.

Note that the only thing that this "configure" script in the top level source directory does is to invoke the configure script in the scripts/ directory, so you could just as easily cd to the scripts directory and run the configure script there.  It will accomplish the same thing.

**Comment by K0rrid0r** [comment]
Line 7 in configure is what you wrote above

![image](https://user-images.githubusercontent.com/56724286/113080185-c6d7ea80-91d6-11eb-9995-c39050b21166.png)

Invoking  ./configure in:
C:\cygwin64\usr\local\src\irsim\scripts
![image](https://user-images.githubusercontent.com/56724286/113081544-2931ea80-91d9-11eb-9bf5-83a402293599.png)

Shows a slightly different error

![image](https://user-images.githubusercontent.com/56724286/113080226-d820f700-91d6-11eb-90be-214fdd3088bc.png)

**Comment by RTimothyEdwards** [comment]
Seems like some kind of issue with DOS-style vs. UNIX-style carriage-return/linefeed characters?

I haven't used cygwin for a very long time.  It was basically made obsolete by WSL.  But on Windows 8.1 you don't have that choice.

**Comment by K0rrid0r** [comment]
hmm, that might be it.

1.

i came a bit further when i converted configure to UNIX style in notepad++

![image](https://user-images.githubusercontent.com/56724286/113083111-e887a080-91db-11eb-9cb3-6d00d7cdabee.png)

maybe i should just convert all those files in scripts

2.
converted config.sub also

![image](https://user-images.githubusercontent.com/56724286/113083462-88452e80-91dc-11eb-9073-f34a7a6d6658.png)

3.
converted all the files in scripts

![image](https://user-images.githubusercontent.com/56724286/113083924-6304f000-91dd-11eb-83b2-41cb1a6122c0.png)

4

adding the logfile also
[config.log](https://github.com/RTimothyEdwards/irsim/files/6233580/config.log)

this seems to be the problem
![image](https://user-images.githubusercontent.com/56724286/113087821-b3338080-91e4-11eb-9926-ecd546ac304d.png)



i see im missing a few -maybe- essential gcc libs, im adding them and see if its works
EDIT: damn that didnt help...

**Comment by K0rrid0r** [comment]
Right some update
I managed to build irsim by replacing the configure file from the one in the magic directory.

only thing left now is this

![image](https://user-images.githubusercontent.com/56724286/113146507-0c74d180-9230-11eb-8398-0564eef215c9.png)

strange because the file is there
![image](https://user-images.githubusercontent.com/56724286/113147298-ef8cce00-9230-11eb-983d-81c323d4cb42.png)

**Comment by RTimothyEdwards** [comment]
The question here is whether there is such a file "/usr/local/lib/irsim/tcl/tclirsim.dll".  The "make" process should have created the DLL file, and "(sudo) make install" should have installed it.

I would expect issues with just using magic's "configure" script, although they are very similar.  You might want to try doing "cd scripts ; aclocal ; autoconf " to generate a new configure script, if nothing else seems to be working.

**Comment by K0rrid0r** [comment]
had to do the following

1.
git clone git://opencircuitdesign.com/irsim

2.
dos2unix the whole dir with:
find . -type f -print0 | xargs -0 dos2unix

3.
create a new configure file with (thanks)
aclocal ; autoconf 

4.
and finally cd to: usr\local\src\irsim\
./configure --disable-usersubckt
make
make install

and i think its correct now, it looks like it at least?

![image](https://user-images.githubusercontent.com/56724286/113153345-4695a180-9237-11eb-9c4b-d1437b7acf2c.png)

(now i just need to learn how to combine it with magic etc)

**Comment by RTimothyEdwards** [comment]
Yes, that looks good.  Also good do know that the build process still works with Cygwin.  Thanks for writing down the instructions that get it all to work right.

**Comment by K0rrid0r** [comment]
more exact install note for win 8.1 requires 'wget' and 'dos2unix' from cygwin package install, if you want you can put these instaructions on the homepage.


launch the Xserver with startxwin

place the file in C:\cygwin64
wget http://opencircuitdesign.com/cygwin/archive/irsim_9_7_92.tgz
tar xfz netgen_1_5_46.tgz

place the git clone in C:\cygwin64\usr\local\src
git clone git://opencircuitdesign.com/irsim
cd irsim
find . -type f -print0 | xargs -0 dos2unix
cd scripts
*rename configure.in to configure.ac and remove the 'configure' file*
aclocal ; autoconf
cd ..
./configure --disable-usersubckt
make
make install
irsim

**Comment by umarcor** [comment]
FTR, using `git config --global core.autocrlf input` might prevent the need for dos2unix.
